build.sbt

```diff
@@ -8,6 +8,8 @@ scalaVersion := "2.11.8"
 libraryDependencies ++= Seq(
     "com.twitter" %% "scrooge-core" % "4.15.0",
     "org.apache.thrift" % "libthrift" % "0.10.0",
+    "com.novus" %% "salat-core" % "1.9.9",
+    "com.github.luben" % "zstd-jni" % "1.3.1-1",
```

Comment:
we should used `1.0.0` for the time being as this is the one we have tested on `CODE`.
We need to check cross versions compatibility before upgrading.

Reply:
There is at least [one open issue on the current version](https://github.com/luben/zstd-jni/issues/44) as well that could affect us

Reply:
Yes, actually I got this error when I tested locally 👍

Reply:
I know this is very old, but interestingly, we didn't notice that this adds [3.6MB](https://mvnrepository.com/artifact/com.github.luben/zstd-jni/1.3.5-2) of mostly binary dlls/shared binaries to the size of the jar.  This is fine in a normal ASG but I noticed this because I've got a (non critical speed) lambda that depends on ophan event model in acquisition event producer which in turn depends on thrift serialiser.
Thrift-serrialiser itself is [just 21kb](https://mvnrepository.com/artifact/com.gu/thrift-serializer_2.12/3.0.0) which is good.
I think part of the reason the minimum cold start for a scala lambda is >5s is because often we end up with a lot of dependencies on the classpath that are only theoretically necessary.  This also has effects on snyk, where we have a lot of vulnerable libraries flagged that are not a problem as they are not called.

I'm not sure what the solution is, other than breaking up dependencies into more granular parts (but still keeping them in the same repo) and minimising their dependencies.  This would end up being like the AWS client libraries where you have to pull in what you want individually.  And even the AWS client pulls in a lot of stuff!  A more practical solution without too much refactoring would be to use some kind of jar stripper to remove unreachable files, but that would have no idea about non-bytecode resources.

I'd be interested in any thoughts (even if it's just that for critical speed lambdas we should use zio-lambda or typescript)

Reply:
since this is always used over HTTP, i'd be inclined to release separate artefacts and use `accept-encoding` to opt-in either gzip, zstandard, or other. right now this library is pretty much post-hoc and there;s a lot of manual wrangling to use it, but it could evolve into okhttp interceptors and some such.

Reply:
Thanks for the reply, although I'm not quite sure what you mean by the second half of that, but I like the separate artefact idea.  The same principle should apply to acquisition event producer.  And then to support-workers.  I will do some refactoring at least down to the acquisition event producer at some convenient point I think, and may take it further into support workers and this lib.